cmd/aries-agentd/startcmd/start.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package startcmd
+
+import (
+	"errors"
+	"fmt"
+	"github.com/gorilla/mux"
+	"github.com/hyperledger/aries-framework-go/pkg/common/log"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries"
+	"github.com/hyperledger/aries-framework-go/pkg/framework/aries/defaults"
+	"github.com/hyperledger/aries-framework-go/pkg/restapi"
+	"github.com/spf13/cobra"
+	"net/http"
+	"strings"
+)
+
+const AgentHostFlagName = "api-host"
+const AgentHostFlagShorthand = "a"
+const AgentHostFlagUsage = "Host Name:Port"
+
+const AgentInboundHostFlagName = "inbound-host"
+const AgentInboundHostFlagShorthand = "i"
+const AgentInboundHostFlagUsage = "Inbound Host Name:Port"
+
+const AgentDBPathFlagName = "db-path"
+const AgentDBPathFlagShorthand = "d"
+const AgentDBPathFlagUsage = "Path to database"
+
+const MissingHostErrorMessage = "Unable to start aries agentd, host not provided"
+const MissingInboundHostErrorMessage = "Unable to start aries agentd, HTTP Inbound transport host not provided"
+
+var logger = log.New("aries-framework/agentd")
+
+type server interface {
+	ListenAndServe(host string, router *mux.Router) error
+}
+
+type RealServer struct {
+	server http.Server
+}
+
+func (s *RealServer) ListenAndServe(host string, router *mux.Router) error {
+	return http.ListenAndServe(host, router)
+}
+
+func Cmd(server server) *cobra.Command {
+	startCmd := &cobra.Command{
+		Use:   "start",
+		Short: "Start an agent",
+		Long:  `Start an Aries agent controller`,
+
+		Run: func(cmd *cobra.Command, args []string) {
+			host, err := cmd.Flags().GetString(AgentHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			inboundHost, err := cmd.Flags().GetString(AgentInboundHostFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			dbPath, err := cmd.Flags().GetString(AgentDBPathFlagName)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+
+			err = startAgent(server, host, inboundHost, dbPath)
+			if err != nil {
+				logger.Errorf(err.Error())
+			}
+		},
+	}
+	startCmd.Flags().StringP(AgentHostFlagName, AgentHostFlagShorthand, "", AgentHostFlagUsage)
+	startCmd.MarkFlagRequired(AgentHostFlagName)
+	startCmd.Flags().StringP(AgentInboundHostFlagName, AgentInboundHostFlagShorthand, "", AgentInboundHostFlagUsage)
+	startCmd.MarkFlagRequired(AgentInboundHostFlagName)
+	startCmd.Flags().StringP(AgentDBPathFlagName, AgentDBPathFlagShorthand, "", AgentDBPathFlagUsage)
+	startCmd.MarkFlagRequired(AgentDBPathFlagName)
+
+	return startCmd
+}
+
+func startAgent(server server, host string, inboundHost string, dbPath string) error {
+	if host == "" {
+		return errors.New(strings.ToLower(MissingHostErrorMessage))
+	}
+
+	var opts []aries.Option
+	if inboundHost == "" {
+		return errors.New(strings.ToLower(MissingInboundHostErrorMessage))
+	}
+	opts = append(opts, defaults.WithInboundHTTPAddr(inboundHost))
```

Comment:
`var opts []aries.Option` should be before line 97

Reply:
This was code that I didn't originally write - I just moved it from another file to here. It was an easy change to make though, so I did it.